src/python-common/setup.py

```diff
@@ -15,6 +16,22 @@
     long_description = fh.read()
 
 
+class PyTest(TestCommand):
+    user_options = [('addopts=', 'a', "Arguments to pass to py.test")]
+
+    def initialize_options(self):
+        TestCommand.initialize_options(self)
+        self.addopts = []
+
+    def run_tests(self):
+        # import here, cause outside the eggs aren't loaded
+        import pytest
+
+        args = self.addopts.split() if isinstance(self.addopts, str) else self.addopts
```

Comment:
i'd suggest avoid using `isinstance()`. instead, i recommend

- normalize the options in `finalize_options()`. 
- and

```py
if self.addopts:
  self.addopts = self.addopts.split()
else:
  self.addopts = []
```

Reply:
In py2, self.addopts is a tuple, in py3, self.addopts is a string.